subprojects/docs/src/samples/build-organization/cross-project-output-sharing/kotlin/producer/build.gradle.kts

```
@@ -0,0 +1,19 @@
+val sharedFile: Provider<RegularFile> = project.layout.buildDirectory.dir("some-subdir").map { it.file("shared-file.txt") }
+
+val makeFile = tasks.register("makeFile") {
+    outputs.file(sharedFile)
+
+    val fileToWrite = sharedFile.get().asFile
+    doFirst {
+        fileToWrite.writeText("This file is shared across Gradle subprojects.")
+    }
+}
+
+val sharedConfiguration: Configuration by configurations.creating {
```

No need for the explicit type declaration, creating a configuration creates a configuration.

```suggestion
val sharedConfiguration by configurations.creating {
```

---

Included initially because IDEA complains otherwise. I'm fine with removing it since you mentioned this isn't something we do in other snippets, and IDEA's warning for this is pretty gentle.